type-definitions/immutable.js.flow

```diff
@@ -526,8 +526,14 @@ declare module 'immutable' {
     updateIn(keyPath: ESIterable<any>, notSetValue: any, value: any): Map<K,V>;
     updateIn(keyPath: ESIterable<any>, value: any): Map<K,V>;
 
-    mergeIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
-    mergeDeepIn(keyPath: ESIterable<any>, ...iterables: ESIterable<any>[]): Map<K,V>;
+    mergeIn(
+      keyPath: ESIterable<any>,
+      ...iterables: (ESIterable<any> | { [key: string]: any })[]
+    ): Map<K,V>;
```

Comment:
Done.

> maybe the key path could be ESIterable<K> to make it type safe?

Agreed, done.

> It should also be possible to merge it with a Map, not only an object, right?

Right, also w/ `List` etc., but I think it's handled by `ESIterable<K>`. Tests added.